### DISCUSSION TITLE
context.query is undefined in getInitialProps

### DISCUSSION THREAD

**Discussion by sondh5** [vercel/next.js#13327] [Help]
Hi, I have a bit confusing about this case below, anyone please help me find out the reason. 

```js
import { useRouter } from 'next/router'
import Layout from '../../components/MyLayout.js'

export default function Post() {
  const router = useRouter()

  return (
    <Layout>
      <h1>{router.query.id}</h1>
      <p>This is the blog post content.</p>
    </Layout>
  )
}

Post.getInitialProps = async function (context) {
  console.log(context.query)
  return { id: context.query.id }
}
```
Source: https://github.com/zeit/next-learn-demo

When I access any url `/p/whatever` route directly in the browser it will return me the id equal `whatever`.
But only with `/p/[id]` I get `undefined` `context.query.id` instead of `[id]`.

Thanks for your reading.

**Comment by lamroger** [comment]
Bump - running into this too. Any help would be appreciated! Thanks in advance!

found a larger discussion: https://github.com/vercel/next.js/discussions/11484
